SonarQube.TeamBuild.PostProcessor/SummaryReportBuilder.cs

```diff
@@ -0,0 +1,164 @@
+                throw new ArgumentNullException(nameof(result));
+            }
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.settings = settings;
+            this.config = config;
+            this.result = result;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Generates summary reports for LegacyTeamBuild and for Build Vnext
+        /// </summary>
+        public static void GenerateReports(TeamBuildSettings settings, AnalysisConfig config, ProjectInfoAnalysisResult result, ILogger logger)
+        {
+            SummaryReportBuilder reportBuilder = new SummaryReportBuilder(settings, config, result, logger);
+            reportBuilder.GenerateReports();
+        }
+
+        private void GenerateReports()
+        {
+            if (this.settings.BuildEnvironment == BuildEnvironment.LegacyTeamBuild
+                && !TeamBuildSettings.SkipLegacyCodeCoverageProcessing)
+            {
+                UpdateTeamBuildSummary();
+            }
+
+            if (this.settings.BuildEnvironment == BuildEnvironment.TeamBuild)
+            {
+                CreateBuildVnextSummaryMdFile();
+            }
+        }
+
+        private void CreateBuildVnextSummaryMdFile()
+        {
+            this.logger.LogMessage(Resources.Report_CreatingSummaryMarkdown);
+
+            int skippedProjectCount, invalidProjectCount, excludedProjectCount, productProjectCount, testProjectCount;
+            GetProjectCountByType(out skippedProjectCount, out invalidProjectCount, out excludedProjectCount, out productProjectCount, out testProjectCount);
+
+            Debug.Assert(!String.IsNullOrEmpty(this.config.SonarOutputDir), "Could not find the output directory");
+            string summaryMdPath = Path.Combine(this.config.SonarOutputDir, "summary.md");
+
+            using (StreamWriter sw = new StreamWriter(summaryMdPath, append: false))
+            {
+                string projectDescription = GetProjectDescription();
+
+                sw.WriteLine(Resources.Report_MdSummaryTitle);
+
+                if (this.config.SonarRunnerPropertiesPath != null && this.result.RanToCompletion)
+                {
+                    string sonarUrl = GetSonarDashboadUrl();
+                    sw.WriteLine(Resources.Report_MdSummaryAnalysisSucceeded, projectDescription, sonarUrl);
+                }
+
+                if (!this.result.RanToCompletion)
+                {
+                    sw.WriteLine(Resources.Report_MdSummaryAnalysisFailed, projectDescription);
+                }
+
+                sw.WriteLine(Resources.Report_MdSummaryProductAndTestMessage, productProjectCount, testProjectCount);
+                sw.WriteLine(Resources.Report_MdSummaryInvalidSkippedAndExcludedMessage, invalidProjectCount, skippedProjectCount, excludedProjectCount);
+            }
+        }
+
+        private void UpdateTeamBuildSummary()
```

Comment:
This method already existed, I just refactored some common functionality with the method above.